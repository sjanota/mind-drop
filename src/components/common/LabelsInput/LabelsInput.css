div.LabelsInput {
    display: flex;
    align-items: center;
    flex-wrap: wrap;
    height: auto;
    padding: 0;
}

.LabelsInput > input {
    width: auto;
    flex-grow: 2;
    border: none;
    background: transparent;
    min-width: 30px;
    padding-left: 0;
}

.LabelsInput .RemovableLabel {
    display: inline-flex;
    align-items: center;
    margin: 11px 6px 11px 3px;
}

.LabelsInput .RemovableLabel > span {
    margin-right: 5px;
}

.LabelsInput .RemovableLabel button {
    border: none;
    background:transparent;
    padding: 0;
    margin: 0;
    color: inherit;
    display: inline-flex;
    align-items: center;
}
