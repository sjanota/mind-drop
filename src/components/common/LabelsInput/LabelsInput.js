import React from 'react';
import './LabelsInput.css';
import Badge from "react-bootstrap/Badge";
import Octicon, {X} from "@primer/octicons-react";
import PropTypes from 'prop-types';

function RemovableLabel({label, onDelete}) {
  return <Badge className={"RemovableLabel"} variant={"primary"}>
    <span>{label}</span>
    <button onClick={onDelete}><Octicon width={8} icon={X}/></button>
  </Badge>
}

export default function LabelsInput({labels, setLabels}) {
  const [inputValue, setInputValue] = React.useState("");

  const onChange = (evt) => {
    const value = evt.target.value;
    if (value.endsWith(" ")) {
      setInputValue("");
      setLabels(old => [...old, value.trim()]);
    } else {
      setInputValue(value);
    }
  };

  const onDeleteLabel = (label) => () => {
    setLabels(old => [...old.slice(0, old.indexOf(label)), ...old.slice(old.indexOf(label) + 1, old.length)])
  };

  return <div className={"form-control LabelsInput"}>
    {labels.map(l => <RemovableLabel label={l} key={l} onDelete={onDeleteLabel(l)}/>)}
    <input type='text' value={inputValue} onChange={onChange} className={"form-control"}/>
  </div>
}

LabelsInput.propTypes = {
  labels: PropTypes.array.isRequired,
  setLabels: PropTypes.func.isRequired
};
