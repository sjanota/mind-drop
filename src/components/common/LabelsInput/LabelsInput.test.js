import React from 'react';
import ReactDOM from 'react-dom';
import LabelsInput from "./LabelsInput";

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<LabelsInput labels={[]} setLabels={() => {}}/>, div);
  ReactDOM.unmountComponentAtNode(div);
});


