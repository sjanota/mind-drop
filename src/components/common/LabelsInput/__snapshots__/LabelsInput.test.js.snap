// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="form-control LabelsInput"
>
  <span
    className="RemovableLabel badge badge-primary"
  >
    <span>
      label1
    </span>
    <button
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="octicon"
        height={10.666666666666666}
        role="img"
        style={
          Object {
            "display": "inline-block",
            "fill": "currentColor",
            "userSelect": "none",
            "verticalAlign": "text-bottom",
          }
        }
        viewBox="0 0 12 16"
        width={8}
      >
        <path
          d="M7.48 8l3.75 3.75-1.48 1.48L6 9.48l-3.75 3.75-1.48-1.48L4.52 8 .77 4.25l1.48-1.48L6 6.52l3.75-3.75 1.48 1.48L7.48 8z"
          fillRule="evenodd"
        />
      </svg>
    </button>
  </span>
  <span
    className="RemovableLabel badge badge-primary"
  >
    <span>
      label2
    </span>
    <button
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="octicon"
        height={10.666666666666666}
        role="img"
        style={
          Object {
            "display": "inline-block",
            "fill": "currentColor",
            "userSelect": "none",
            "verticalAlign": "text-bottom",
          }
        }
        viewBox="0 0 12 16"
        width={8}
      >
        <path
          d="M7.48 8l3.75 3.75-1.48 1.48L6 9.48l-3.75 3.75-1.48-1.48L4.52 8 .77 4.25l1.48-1.48L6 6.52l3.75-3.75 1.48 1.48L7.48 8z"
          fillRule="evenodd"
        />
      </svg>
    </button>
  </span>
  <span
    className="RemovableLabel badge badge-primary"
  >
    <span>
      label3
    </span>
    <button
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="octicon"
        height={10.666666666666666}
        role="img"
        style={
          Object {
            "display": "inline-block",
            "fill": "currentColor",
            "userSelect": "none",
            "verticalAlign": "text-bottom",
          }
        }
        viewBox="0 0 12 16"
        width={8}
      >
        <path
          d="M7.48 8l3.75 3.75-1.48 1.48L6 9.48l-3.75 3.75-1.48-1.48L4.52 8 .77 4.25l1.48-1.48L6 6.52l3.75-3.75 1.48 1.48L7.48 8z"
          fillRule="evenodd"
        />
      </svg>
    </button>
  </span>
  <input
    className="form-control"
    onChange={[Function]}
    type="text"
    value=""
  />
</div>
`;
